### ISSUE TITLE
sort inventory variable imported from source

### ISSUE THREAD

**Issue by hudecof** [ansible/awx#252]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
 - UI

##### SUMMARY
When importing invetroy from source, in my case project source, the host variables are in random order. It eould be nice to sort then.
If you obey the rule for roles variables with prefix, al role variables then would be in one place of the list.

##### ENVIRONMENT
* AWX version: e6777577b6b75e1f3e1b064c12432af3101c1f53
* Ansible version:  2.4.0

##### STEPS TO REPRODUCE
Import the inventory from any extrenal source, in my case the result for host is
```
ansible_ssh_host: 192.168.93.101
resolv_options:
  - 'timeout:30'
  - 'attempts:5'
  - rotate
resolv_nameserver:
  - 192.168.80.11
  - 192.168.80.12
resolv_domain: example.com
resolv_search:
  - example.com
ansible_ssh_user: root
```

##### EXPECTED RESULTS
```
ansible_ssh_host: 192.168.93.101
ansible_ssh_user: root
resolv_domain: example.com
resolv_nameserver:
  - 192.168.80.11
  - 192.168.80.12
resolv_options:
  - 'timeout:30'
  - 'attempts:5'
  - rotate
resolv_search:
  - example.com
```

##### ACTUAL RESULTS

the variables dictionary is unsorted

##### ADDITIONAL INFORMATION

One more question.Why are you merging the group variables into the host? In my examke the variables **resolv_options** is in the file **group_vars/all**

**Comment by matburt** [comment]
This is more of an aesthetic thing, but I can understand.

> Why are you merging the group variables into the host?

This has to do with the way `ansible-inventory` (or the backport thereof) works currently. I think on the core side they are working on changing it. We've moved to using this for importing inventory because of some architectural changes coming up in Ansible itself.

**Comment by hudecof** [comment]
thanks, this is just nice to have

**Comment by jlmitch5** [comment]
@matburt as far as I understand it, we just display the vars we get back, so I think this would need to be done on the api side.  feel free to move back to our bucket if I'm misunderstanding!

**Comment by hudecof** [comment]
@matburt  more les is aesthetic, but could save a lot of time ;)

**Comment by matburt** [comment]
I think the UI ends up rendering what we store/send and possibly causes the reordering. We'll have to investigate this more (or some other enterprising individual who cares a lot)

**Comment by jlmitch5** [comment]
hey @hudecof,
@matburt and I discussed this a bit more and got on the same page.  It _is_ a UI thing...and we do have the ability to set a sorting config variable in the library we use to sort these values (thanks @gconsidine for the tip on that one -- https://www.npmjs.com/package/js-yaml#safedump-object---options-)

this started a larger discussion as to what the correct behavior is in terms of sorting.  Should this (and all) yaml blobs in the UI be sorted based on key?  or should they be sorted in a way that preserves the order in which the user entered (I don't really believe this is relevant in this particular case--I think that yaml is generated through whatever the inventory import process is).

We came to the conclusion that this is subjective, and it's difficult to decide what every user would expect and want to see.  Really appreciate you filing the issue (allowed us to learn about js-yaml's sorting configuration options),  but for now we think the UI should continue to show the yaml without some sort of sorting mechanism until we can come to a (consistent-across-the-app) conclusion as to what that should be.
